app/src/main/java/com/android/shelfLife/ui/newutils/NewFoodItemDetail.kt

```diff
@@ -0,0 +1,104 @@
+            ?: stringResource(R.string.food_item_not_opened)
+    val formattedBuyDate = foodItem.buyDate?.let { formatTimestampToDisplayDate(it) }
+
+    ElevatedCard(
+        modifier =
+        Modifier.fillMaxWidth()
+            .padding(horizontal = 16.dp, vertical = 8.dp)
+            .testTag("foodItemDetailsCard")) {
+        Column(modifier = Modifier.padding(16.dp)) {
+            FoodItemDetailText(
+                text =
+                stringResource(
+                    R.string.food_item_category_label, foodItem.foodFacts.category.name),
+                tag = "categoryText",
+                style = textStyle)
+            FoodItemDetailText(
+                text = stringResource(R.string.food_item_location_label, foodItem.location.name),
+                tag = "locationText",
+                style = textStyle)
+            FoodItemDetailText(
+                text = stringResource(R.string.food_item_status_label, foodItem.status.name),
+                tag = "statusText",
+                style = textStyle)
+            FoodItemDetailText(
+                text = stringResource(R.string.food_item_expiry_date_label, formattedExpiryDate),
+                tag = "expiryDateText",
+                style = textStyle)
+            FoodItemDetailText(
+                text = stringResource(R.string.food_item_open_date_label, formattedOpenDate),
+                tag = "openDateText",
+                style = textStyle)
+            formattedBuyDate?.let { stringResource(R.string.food_item_buy_date_label, it) }?.let {
+                FoodItemDetailText(
+                    text = it,
+                    tag = "buyDateText",
+                    style = textStyle)
+            }
+            FoodItemDetailText(
+                text =
+                stringResource(
+                    R.string.food_item_energy_label,
+                    foodItem.foodFacts.nutritionFacts.energyKcal),
+                tag = "energyText",
+                style = textStyle)
+            FoodItemDetailText(
+                text =
+                stringResource(
+                    R.string.food_item_proteins_label,
+                    foodItem.foodFacts.nutritionFacts.proteins),
+                tag = "proteinsText",
+                style = textStyle)
+            FoodItemDetailText(
+                text =
+                stringResource(
+                    R.string.food_item_quantity_label,
+                    foodItem.foodFacts.quantity.amount,
+                    foodItem.foodFacts.quantity.unit.name),
+                tag = "quantityText",
+                style = textStyle)
+        }
+    }
+}
+
+/**
+ * Composable function to display a text detail of a food item.
+ *
+ * @param text The text to be displayed.
+ * @param tag The test tag for the text.
+ * @param style The style to be applied to the text.
+ */
+@Composable
+fun FoodItemDetailText(text: String, tag: String, style: TextStyle) {
```

Comment:
You have moved this functions from the utils to this file as it was not being used anywhere else. This is good, it makes the code easier to read.